### DISCUSSION TITLE
2022-04-04 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#1259] [Maintainer discussions]
## About
At the start of each week, we start a discussion covering Mathesar's progress, team member check ins, and general updates. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

## Discussion topics
1. [Last Week in Mathesar](https://github.com/centerofci/mathesar/discussions/1259#discussioncomment-2493616)
2. [Frontend priorities](https://github.com/centerofci/mathesar/discussions/1259#discussioncomment-2493617)
3. [Backend priorities](https://github.com/centerofci/mathesar/discussions/1259#discussioncomment-2493618)
4. [Design priorities](https://github.com/centerofci/mathesar/discussions/1259#discussioncomment-2493619)
5. [Product priorities](https://github.com/centerofci/mathesar/discussions/1259#discussioncomment-2493620)
6. [Comms Assignee schedule](https://github.com/centerofci/mathesar/discussions/1259#discussioncomment-2493621)
7. [Sean's check in](https://github.com/centerofci/mathesar/discussions/1259#discussioncomment-2499794)
8. [Pavish's check in](https://github.com/centerofci/mathesar/discussions/1259#discussioncomment-2502935)
9. [Kriti's check-in](https://github.com/centerofci/mathesar/discussions/1259#discussioncomment-2503470)
10. [Ghislaine's check-in](https://github.com/centerofci/mathesar/discussions/1259#discussioncomment-2503772)
11. [Mukesh's check-in](https://github.com/centerofci/mathesar/discussions/1259#discussioncomment-2507423)

## Weekly check-in
@centerofci/mathesar-core please post your weekly check-in as a top-level response and edit the discussion topics above to add your check in as a topic. Others are also welcome to respond.

Prompts:
- How did your most recent work go?
- What are you planning to work on soon?
- Do you anticipate any dependencies or blockers that someone else on the team can help you with?

If you'd like to add anything else to your check-in that's not related to the prompts, feel free to do so.

**Comment by kgodey** [comment]
## Last Week in Mathesar

### Feedback Requested
- https://github.com/centerofci/mathesar-wiki/pull/42
- https://github.com/centerofci/mathesar/discussions/1243

### Complete
- **Milestones**:
    - [2022-03 improvements](https://github.com/centerofci/mathesar/milestone/44?closed=1) with 28 issues completed
- **Discussions**:
    - https://github.com/centerofci/mathesar/discussions/1242
- **Full stack and general work**:
    - https://github.com/centerofci/mathesar/issues/1131
- **Backend work**:
    - https://github.com/centerofci/mathesar/issues/876
- **Frontend work**:
    - https://github.com/centerofci/mathesar/issues/906
    - https://github.com/centerofci/mathesar/issues/1099
    - https://github.com/centerofci/mathesar/issues/970
    - https://github.com/centerofci/mathesar/issues/927
    - https://github.com/centerofci/mathesar/pull/1236
    - https://github.com/centerofci/mathesar/pull/1239
- **Closed without work**:
    - https://github.com/centerofci/mathesar/issues/224
    - https://github.com/centerofci/mathesar/issues/223
    - https://github.com/centerofci/mathesar/issues/401
    - https://github.com/centerofci/mathesar/issues/440
    - https://github.com/centerofci/mathesar/issues/396
    - https://github.com/centerofci/mathesar/issues/395
    - https://github.com/centerofci/mathesar/issues/441
    - https://github.com/centerofci/mathesar/issues/1245
    - https://github.com/centerofci/mathesar/issues/1234
    - https://github.com/centerofci/mathesar/issues/1246

**Comment by kgodey** [comment]
## Frontend priorities

1. Get open PRs merged:
    1. https://github.com/centerofci/mathesar/pull/1182
    1. https://github.com/centerofci/mathesar/pull/1211
    1. https://github.com/centerofci/mathesar/pull/1225
    1. https://github.com/centerofci/mathesar/pull/1241
    1. https://github.com/centerofci/mathesar/pull/1254
    1. https://github.com/centerofci/mathesar/pull/1256
    1. https://github.com/centerofci/mathesar/pull/1257
2. Make progress on high-priority issues for [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) milestone:
    1. #1108
    1. #257 
    1. #775
3. Continue working on other [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) frontend issues.

@pavish @seancolsen Please comment below with thoughts/concerns.

**Reply by seancolsen** [reply]
Looks good

**Reply by pavish** [reply]
Looks good to me

**Comment by kgodey** [comment]
## Backend priorities

1. Get open PRs completed and merged:
    1. https://github.com/centerofci/mathesar/pull/1205
    1. https://github.com/centerofci/mathesar/pull/1251
    1. https://github.com/centerofci/mathesar/pull/1255
2. Complete remaining backend work for [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11):
    1. @mathemancer
        1. https://github.com/centerofci/mathesar/issues/408 
        2. https://github.com/centerofci/mathesar/issues/411 
        3. https://github.com/centerofci/mathesar/issues/414
        4. https://github.com/centerofci/mathesar/issues/384
        5. https://github.com/centerofci/mathesar/issues/429
        6. https://github.com/centerofci/mathesar/issues/419
        7. https://github.com/centerofci/mathesar/issues/432
    1. @silentninja:
        1. https://github.com/centerofci/mathesar/issues/1105
        1. https://github.com/centerofci/mathesar/issues/1228
    1. Not assigned to core team members:
        1. https://github.com/centerofci/mathesar/issues/1145 (unassigned)
        1. https://github.com/centerofci/mathesar/issues/1167 (assigned to @ratika-12)
        1. https://github.com/centerofci/mathesar/issues/1214 (assigned to @Cronus1007)
3. Start work on backend work for [Links between Tables](https://github.com/centerofci/mathesar/milestone/47), some ideas:
    1. https://github.com/centerofci/mathesar/issues/943
    2. https://github.com/centerofci/mathesar/issues/944
    3. https://github.com/centerofci/mathesar/issues/973
    4. https://github.com/centerofci/mathesar/issues/972
4. #1008
5. These issues are on hold until @dmos62 is back from vacation.
    1. https://github.com/centerofci/mathesar/pull/1174
    1. https://github.com/centerofci/mathesar/pull/1222
    1. https://github.com/centerofci/mathesar/issues/1100
    1. https://github.com/centerofci/mathesar/issues/1037

@mathemancer @silentninja Please comment below with thoughts/concerns.

**Comment by kgodey** [comment]
## Design priorities

1. Complete one round of review for #1065
2. Submit first draft of spec for review for:
    1. https://github.com/centerofci/mathesar/issues/466
    2. https://github.com/centerofci/mathesar/issues/978

@ghislaineguerin @ppii775 Please comment below with thoughts/concerns.

**Reply by ghislaineguerin** [reply]
@kgodey looks good

**Comment by kgodey** [comment]
## Product priorities

1. Start research and documentation of Mathesar target use cases for alpha release. 
1. Start figuring out goals for UI styling milestone.

**Comment by kgodey** [comment]
## Comms Assignee schedule

| Week | Person |
|-|-|
| 2022-04-04 to 2022-04-08 | Ghislaine  |
| 2022-04-11 to 2022-04-12 | Kriti |
| 2022-04-13 to 2022-04-15 | Dominykas |
| 2022-04-18 to 2022-04-22 | Brent |
| 2022-04-25 to 2022-04-29 | Kriti |
| 2022-05-02 to 2022-05-06 | Pavish |
| 2022-05-09 to 2022-05-13 | Mukesh |
| 2022-05-16 to 2022-05-20 | Sean |

### Instructions
You can also find the schedule and instructions for the Comms Assignee on the [wiki page](https://wiki.mathesar.org/team/guide/comms-assignee). Please feel free to update the wiki page if you think of anything useful during your shift.

**Comment by seancolsen** [comment]
## Sean's check in

> How did your most recent work go?

- Unfortunately I did zero coding last week. I was out for three days due to a horrendous stomach bug which made 3/4 of my family very sick. The two days I _did_ work were entirely consumed by discussions, reviews, and comms.

> What are you planning to work on soon?

- #775 is still my highest priority. It may require some more refactoring. I'm not sure. I need to get my head back into the problem before I have a good sense of how long I expect it to take. I'm really hoping I can make some headway on this, though I do have a number of smaller discussion/review/comms tasks queued at the moment.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

- No

**Comment by pavish** [comment]
## Pavish's check in

> How did your most recent work go?

Much slower than planned. I was focused more on reviews, discussions and figuring out architecture during the week, and planned to get to focused coding during the weekend (when I'm not distracted by anything else), which did not work out due to personal issues.

> What are you planning to work on soon?

My plan stays the same as last week, except that a lot of figuring out part is done. I'll have to do the grunt coding part.
* https://github.com/centerofci/mathesar/issues/1250
* https://github.com/centerofci/mathesar/issues/257

I anticipate more discussions and PR reviews this week as well.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None

**Comment by kgodey** [comment]
> How did your most recent work go?

I didn't do much last week other than meetings and keeping up with notifications because I was dealing with some personal issues. I have been reading books about design and copy, which have been helpful in thinking through future work.

> What are you planning to work on soon?

1. Support GSoC as needed – proposal reviews, mentoring guidance, etc.
1. Write job descriptions for new roles and get the hiring process started
1. Start research and documentation of use cases and user personas for Mathesar's alpha releas. I'm prioritizing this to:
    1. help us create concrete goals for our launch preparation milestones (UI Styling, Deployment, etc.)
    1. get the team on the same page about product short-term and long-term goals, it should help with discussions like #1243 
1. Think through how the team can collaborate better on the Working with Views milestone, since it will be big and highly interconnected.
1. Still on my plate, but not a priority this week:
    1. #1008 
    2. https://github.com/centerofci/mathesar-wiki/issues/41

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

No.

**Comment by ghislaineguerin** [comment]
> How did your most recent work go?

Really good. Design for views runs smoothly after initial data explorer definition

> What are you planning to work on soon?

Data explorer spec round and spec updates
Design for #466

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None

**Comment by silentninja** [comment]
## Mukesh's check in

> How did your most recent work go?

Quite Slower than I expected as I got held up with some personal issues. 

> What are you planning to work on soon?

I have started working on issues in the `Links between Tables` milestone, planning to continue the progress with it. 

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None

> Personal updates

I will be relocating in next few days, so I am anticipating some distractions and contributions will be slower than usual.

**Comment by mathemancer** [comment]
> * How did your most recent work go?

Things are going well. I'm getting some programming done, but spending plenty of time fielding questions, going through PRs, and participating in discussions.

> * What are you planning to work on soon?

I'm currently trying to figure out why my grouping PR broke seemingly unrelated e2e tests. I suppose that's why the tests are there, since I didn't expect the behaviors indicated to change. After that, I have one quick/easy case to finish up for numeric grouping (DB layer), and then I want to get character grouping done. I expect the character grouping to be easier than numeric, and be easier to reuse for multiple character-like types.

I'm going to wait until I have a few more grouping functions done before adding them to the "descriptive" part of the API, in the hopes that I'll be able to make the descriptions more useful with more of the grouping differences and commonalities in mind.

> * Do you anticipate any dependencies or blockers that someone else on the team can help you with?

Not at the moment.
